contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,315 @@
+# Core Metrics Pipeline in kubelet
+
+**Author**: David Ashpole (@dashpole)
+
+**Last Updated**: 12/21/2016
+
+**Status**: Draft Proposal (WIP)
+
+This document proposes a design for an internal Core Metrics Pipeline.
+
+<!-- BEGIN MUNGE: GENERATED_TOC -->
+
+- [Core Metrics Pipeline in kubelet](#core-metrics-pipeline-in-kubelet)
+  - [Introduction](#introduction)
+    - [Background](#background)
+    - [Motivations](#motivations)
+    - [Proposal](#proposal)
+    - [Non Goals](#non-goals)
+  - [Design](#design)
+    - [Proposed Core Metrics API:](#proposed-core-metrics-api)
+    - [Core Machine Info:](#core-machine-info)
+  - [Implementation Plan](#implementation-plan)
+  - [Rollout Plan](#rollout-plan)
+  - [Implementation Status](#implementation-status)
+
+<!-- END MUNGE: GENERATED_TOC -->
+
+## Introduction
+
+### Background
+The [Monitoring Architecture](https://github.com/kubernetes/kubernetes/blob/master/docs/design/monitoring_architecture.md) proposal contains a blueprint for a set of metrics referred to as "Core Metrics".  The purpose of this proposal is to specify what those metrics are, and how they will be collected on the node.
```

Comment:
I think the argument is that the name "core metrics" is wrong, or not philosophically aligned with our general Kubernetes wide naming conventions (wearing my "what are we trying to build as a project" hat momentarily).

I have badgered Derek a bit about this because "core" isn't a great word to describe things.  I was looking for a more precise name for this general approach, and in particular the name that this API group will have to align well to what it is surfacing.  "metrics" is not appropriate (this is not generic metrics).  "core-metrics" is not precise (it does not define what is core).

"resource-metrics", "system-resource-metrics", "utilization-metrics" are all more accurate.  Before we merge this, I'd like to converge on a name that is not "core" to describe this subset of all possible metrics.